Framework/PythonInterface/test/python/plugins/algorithms/CreatePoleFigureTableWorkspaceTest.py

```diff
@@ -0,0 +1,305 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2025 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import unittest
+from mantid.simpleapi import (
+    CreatePoleFigureTableWorkspace,
+    AnalysisDataService,
+    AlgorithmManager,
+    SetSample,
+    SetGoniometer,
+    CreateWorkspace,
+    EditInstrumentGeometry,
+    CloneWorkspace,
+    CreateEmptyTableWorkspace,
+)
+from mantid.geometry import CrystalStructure
+import numpy as np
+
+
+class CreatePoleFigureTableTest(unittest.TestCase):
+    def setUp(self):
+        ws = CreateWorkspace(DataX=[0.0, 1.0], DataY=[1.0, 2.0, 3.0, 4.0], NSpec=4)
+
+        # set the instrument to have four detectors
+        EditInstrumentGeometry(
+            Workspace="ws",
+            PrimaryFlightPath=50,
+            L2=[50, 50, 50, 50],
+            Polar=[85, 90, 95, 90],
+            Azimuthal=[0, 0, 0, 45],
+            DetectorIDs=[0, 1, 2, 3],
+        )
+        # polar is taken from z towards positive x. Here there are 2 detectors along x axis and 2 at +/-5degs
+        # azimuthal is taken from z as well. Here 3 in XZ plane, 1 at +45 degrees towards Y
+
+        xtal = CrystalStructure("5.431 5.431 5.431", "F d -3 m", "Si 0 0 0 1.0 0.05")
+
+        sample_shape = """
+        <cylinder id="A">
+          <centre-of-bottom-base x="0" y="0" z="0.05" />
+          <axis x="0" y="0" z="-1" />
+          <radius val="0.1" />
+          <height val="0.1" />
+        </cylinder>
+        """
+
+        ws2 = CloneWorkspace(ws)
+
+        SetGoniometer("ws2", Axis0="0,0,1,0,1")
+        SetSample("ws", Geometry={"Shape": "CSG", "Value": sample_shape})
+        ws.sample().setCrystalStructure(xtal)
+
+        SetGoniometer("ws2", Axis0="45,0,1,0,1")
+        SetSample("ws2", Geometry={"Shape": "CSG", "Value": sample_shape})
+        ws2.sample().setCrystalStructure(xtal)
+
+        peak_param_table = CreateEmptyTableWorkspace(OutputWorkspace="PeakParameterWS")
+        for col in ("I", "X0", "chi2"):
+            peak_param_table.addColumn("double", col)
+        for i in range(4):
+            val = 1.0 + (i / 10)
+            peak_param_table.addRow([val, 3.14 + (i / 100), i])  # 3.14 hkl (1,1,1) dSpacing
+
+    def tearDown(self):
+        AnalysisDataService.clear()
+
+    def eval_arrays(self, arr1, arr2, thresh=0.001):
+        self.assertTrue(np.all(np.abs(arr1 - arr2) < thresh))
+
+    def setup_missing_column(self, column_name):
+        # override PeakParameter Table
+        peak_param_table = AnalysisDataService.retrieve("PeakParameterWS")
+        peak_param_table.removeColumn(column_name)
+
+        alg = AlgorithmManager.create("CreatePoleFigureTableWorkspace")
```

Comment:
There are quite a few calls to `alg = AlgorithmManager.create("CreatePoleFigureTableWorkspace")` and then adding various input parameters. Would it be possible to have this in a helper function that just takes `**kwargs` a bit like `exec_child_alg`?